index.js

```diff
@@ -196,48 +199,49 @@ AFRAME.registerComponent('teleport-controls', {
 
       // Set default status as non-hit
       this.teleportEntity.setAttribute('visible', true);
-      this.line.material.color.set(this.curveMissColor);
-      this.line.material.opacity = this.data.missOpacity;
-      this.line.material.transparent = this.data.missOpacity < 1;
       this.hitEntity.setAttribute('visible', false);
       this.hit = false;
+      this.collidedIndex = -1;
+      this.setLineMaterial(this.hit);
 
       if (this.data.type === 'parabolic') {
         v0.copy(direction).multiplyScalar(this.data.curveShootingSpeed);
-
         this.lastDrawnIndex = 0;
-        const numPoints = this.data.drawIncrementally ? this.numActivePoints : this.line.numPoints;
-        for (var i = 0; i < numPoints+1; i++) {
-          var t;
-          if (i == Math.floor(numPoints+1)){
-            t =  numPoints / (this.line.numPoints - 1);
-          }
-          else {
-            t = i / (this.line.numPoints - 1);
-          }
+        // const numPoints = this.data.drawIncrementally ? this.numActivePoints : this.line.numPoints;
```

Comment:
This was here because some users of teleport-controls (not us) may not want the parabola to shoot out gradually, preferring for it to appear instantly and be able to teleport instantly too.